src/main/scala/mesosphere/marathon/api/v2/Validation.scala

```diff
@@ -299,10 +263,42 @@ trait Validation {
     )
 
   def validateAll[T](x: T, all: Validator[T]*): Result = all.map(v => validate(x)(v)).fold(Success)(_ and _)
+
+  def allViolations(result: Result): Seq[ConstraintViolation] = {
+    def renderPath(desc: Description): String = desc match {
+      case Explicit(s) => s"/${s}"
+      case Generic(s) => s"/${s}"
+      case Indexed(index) => s"($index)"
+      case _ => ""
+    }
+    def mkPath(path: Path): String =
+      if (path.isEmpty)
+        "/"
+      else
+        path.map(renderPath).mkString("")
+
+    def collectViolation(violation: Violation, parents: Path): Seq[ConstraintViolation] = {
+      violation match {
+        case RuleViolation(_, constraint, path) => Seq(ConstraintViolation(mkPath(parents ++ path), constraint))
+        case GroupViolation(_, _, children, path) => children.to[Seq].flatMap(collectViolation(_, parents ++ path))
+      }
+    }
+    result match {
+      case Success => Seq.empty
+      case Failure(violations) =>
+        violations.to[Seq].flatMap(collectViolation(_, Nil))
+    }
+  }
 }
 
 object Validation extends Validation {
 
+  case class ConstraintViolation(path: String, constraint: String)
+
+  implicit class AsDescription(val optional: Option[String]) extends AnyVal {
+    def asDescription: Description = ??? //optional.fold[Description](Empty)(Explicit)
```

Comment:
Is `???` the right choice here?

Reply:
Lol, crazy! The tests all pass with this so looks like we don't use this method. I overlooked this when I was doing the effort of getting the original wix 0.6 changes to build.

I'll just remove it.

Reply:
✅